### DISCUSSION TITLE
How to configure prometheus.prometheusSpec.serviceMonitorNamespaceSelector:

### DISCUSSION THREAD

**Discussion by ghost** [prometheus-operator/kube-prometheus#903] [Q&A]
Let's assume I have installed kube-stack-promethes with the full stack of Promethues-Operator/Prometheus/Grafana in the prometheus-operator namespace.

I have a second kube-stack-prometheus that will live and monitor only the things in that namespace...Let's call it `foo`where the only thing needed is Prometheus/Grafana using https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#multiple-releases.

This is working as expected.  In `foo` I have the `prometheus.prometheusSpec.serviceMonitorNamespaceSelector: foo` which is working as expected.

Is there some way in the primary Prometheus other than listing each and every other labeled service-monitor and essentially say `Look at all namespaces except foo`?  If someone has a lot of namespaces it could become somewhat of a tedious process to add each.

I was hoping for pseudocode logic:
```
...
      serviceMonitorNamespaceSelector:
        matchLabels:
          name: !foo # Everything but foo
```

**Comment by s-urbaniak** [comment]
prometheus-operator supports a `--deny-namespaces` flag (see https://github.com/prometheus-operator/prometheus-operator/blob/9f18c1abac9e0331da511f61cc697e6e97c9aef9/cmd/operator/main.go#L179) which reconciles custom resources in all namespaces except the namespaces you specified in the deny list.
